### PR TITLE
WIP: shift to pybids, rework preproc, add topup

### PR DESCRIPTION
**Pull request by mgxd**
- [x] replace info/data source with `layout.get()`
- [x] missing node connections due to info/data source removal
- [x] add topup workflow 
- [x] fix extract main vs extract opp
- [x] datasink and substitutions
### Update
- [x] update preprocessing/registration
- [x] make preprocess / topup run driven
- [x] calc_median per run
- [x] missing stats after bandpass/smoothing
- [x] compare stats between new and old
- [x] ensure non-topup version runs
### Even more update
- [ ] Add [AFQ](https://github.com/yeatmanlab/AFQ) to diffusion pipeline
- [ ] Refactor modeling to use BIDS *_events.tsv files
- [ ] Optimize for running on openmind
- [ ] Write out CIFTI file for resting state

### CODE REVIEW

File: subject_level/fmri_ants_bids.py [hunk 1]
```diff
@@ -1828,6 +1836,11 @@ def extract_subrois(timeseries_file, label_file, indices):
                         help="Activate nipype debug mode" + defstr)
     parser.add_argument('--cc', action="store_true",
                         help="CompCor correction for task analysis" + defstr)
+    parser.add_argument('--sparse', action="store_true",
+    					help="Sparse model specification" + defstr)
+    parser.add_argument('-ta', dest="TA", default=None, type=float,
```
Comment (satra) [inline subject_level/fmri_ants_bids.py]: i think you can add a group here.

**Comment by satra** [comment]
for the diffusion workflow can we add an AFQ (https://github.com/yeatmanlab/AFQ) step? 

@arokem - is there a dipy version of AFQ yet? or should we just wrap the matlab version for now?

**Comment by arokem** [comment]
There's WIP on this here: http://yeatmanlab.github.io/pyAFQ/

If you are comfortable using pre-release software, you can follow this example to roll your own: 

http://yeatmanlab.github.io/pyAFQ/auto_examples/plot_tract_profile.html#sphx-glr-auto-examples-plot-tract-profile-py

But the first line on the front page of the documentation sums the current state of affairs. For now, we are still recommending people use the Matlab version, until we've had a chance to complete the work on this.

File: dwi/dif_eddy_trac_bids.py [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+                 name='trac-prep')
+    
+    def run_bedpost(sid, tracula_dir, dwi_file):
+        """ """
+        import os
+        import shutil
+        from nipype.interfaces.base import CommandLine
+        pwd = os.getcwd()
+        os.chdir(os.path.join(tracula_dir, sid))
+        bedpost = CommandLine('bedpostx_gpu dmri -NJOBS 1 --model=2 --rician', terminal_output='allatonce')
```
Comment (satra) [inline dwi/dif_eddy_trac_bids.py]: note that these parameters may need to be modified depending on whether we are processing prisma or trio data.

Comment (mgxd) [reply dwi/dif_eddy_trac_bids.py]: I can add a `scanner` argument that sets recommended parameters for either trio/prisma, what would you recommend for the prisma data?

**Comment by mgxd** [comment]
this has grown stale now with fmriprep - we would ideally want to use that output and model on top of it

**Pull request closed without merging**